File: src_minhaz/friendsByAge.py
from pyspark import SparkConf, SparkContext

conf = SparkConf().setMaster("local").setAppName("RatingsHistogram")
sc = SparkContext(conf=conf)


def parseLine(line):
    fields = line.split(',')
    return int(fields[2]), int(fields[3])


lines = sc.textFile("./fakefriends.csv")
values = lines.map(parseLine).mapValues(lambda x: (x, 1))
totalsByAge = values.reduceByKey(lambda x, y: (x[0]+y[0], x[1]+y[1]))
avgByAge = totalsByAge.mapValues(lambda x: x[0]/x[1])
results = avgByAge.collect()
for result in results:
    print(result)

File: src/scala/FriendsByAge.scala
import org.apache.spark.SparkContext
import org.apache.spark.SparkConf

object FriendsByAge{
  def main(args: Array[String]): Unit = {
    val conf:SparkConf = new SparkConf().setAppName("Friends By Age").setMaster("local")
    val sc:SparkContext = new SparkContext(conf)

    val textfile = sc.textFile("./fakefriends.csv")
    val values = textfile.map(parseLine).mapValues(x => (x, 1))
    val totalsByAge = values.reduceByKey((x, y) => (x._1+y._1, x._2+y._2))
    val avgByAge = totalsByAge.mapValues(x => (x._1/x._2).toInt).sortByKey()
    val results = avgByAge.collect()
    for (result <- results){
      println(result)
    }
  }

  def parseLine(line: String): (Int, Int) = {
    val fields = line.split(",")
    return (fields(2).toInt, fields(3).toInt)
  }
}
